pkg/plugins/pg/source/source_integration_test.go

```diff
@@ -517,6 +517,49 @@ func TestRead_Iterator(t *testing.T) {
 	assert.True(t, src.sub == nil, "failed to cleanup subscription")
 }
 
+// TestSnapshotIdempotentTeardown asserts that Teardown can be called
+// idempotently in snapshot mode. Since the snapshot should not be finished,
+// we call it immediately and thus the snapshot interrupt error is expected.
```

Comment:
I might be missing something, but why does Teardown() have to be idempotent? I do, however, see that it should return the correct error.

Reply:
There was a discussion @lovromazgon and I had about making Teardown calls idempotent. It will be important when we go to cluster conduit as well.